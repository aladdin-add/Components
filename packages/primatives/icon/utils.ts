import lazy from "react-lazy-named";

export const getIcon = (config, name) => {
  if (config.iconDir) {
    switch (config.iconDir) {
      case "feather":
      default:
        return lazy(() => import("react-feather"), name);
    }
  }

  throw new Error("Unknown icon lib");
};
