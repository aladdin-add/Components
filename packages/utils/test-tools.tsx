import React from "react";
import { mount, shallow } from "enzyme";
import { ThemeProvider } from "styled-components";
import { ConfigContext } from "@//utils/context/config";
import theme from "@//theme";
const config =  require("../../orchard.theme.config.json");

const removeProperties = () => {
  const keys = ["theme", "styledTheme"];
  return {
    test: (val) => {
      return (
        val &&
        typeof val === "object" &&
        "props" in val &&
        Object.keys(val.props).some((prop) => keys.some((key) => key === prop))
      );
    },
    print: (val, serialize) => {
      keys.forEach((key) => {
        delete val.props[key];
      });
      return serialize(val);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors.common,
    modes: theme.colors.modes,
    ...theme.colors.modes.light,
  },
};

export const shallowWithTheme = (children) => shallow(
      <ThemeProvider theme={formattedTheme}>
        {children}
      </ThemeProvider>
  )
    .dive()
    .shallow();

export const mountWithTheme = (children) => {
  expect.addSnapshotSerializer(removeProperties());

  return mount(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>
        {children}
      </ThemeProvider>
    </ConfigContext.Provider>
  );
};
