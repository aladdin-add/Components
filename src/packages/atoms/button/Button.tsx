import React from 'react';
import { InteractableProps } from '@/primatives/interactable';
import Text from '@/primatives/text';

import { StyledInteractable } from './styled';

export interface Props extends InteractableProps {
  iconName?: string;
  variant?: string;
}

const Button = ({
  className,
  children,
  iconName,
  autoid = '',
  variant = 'default',
  ...rest
}: Props) => {
  return (
    <StyledInteractable
      className={className}
      forwardedAs="button"
      autoid={`${autoid || children}_button`}
      typography="button"
      variant={variant}
      {...rest}
    >
      {typeof children === 'string' ? <Text>{children}</Text> : { children }}
    </StyledInteractable>
  );
};

export default Button;
