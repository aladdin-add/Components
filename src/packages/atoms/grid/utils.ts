import { GridProps } from '@aw-web-design/styled-system';

export const getTemplate = (
  template: GridProps['gridTemplateColumns'] | GridProps['gridTemplateRows'],
) => {
  if (Array.isArray(template)) {
    return template.map((value) => (typeof value === 'number' ? `repeat(${template},1fr)` : value));
  }
  return typeof template === 'number' ? `repeat(${template},1fr)` : template;
};

export const getGrid = (edge: number, value: number): string | number => {
  if (edge && value) {
    return `${edge} / span ${value}`;
  } if (value) {
    return `span ${value}`;
  }
  return edge;
};
