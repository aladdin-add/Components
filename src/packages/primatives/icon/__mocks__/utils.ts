import { GitHub } from 'react-feather';

export const getIcon = () => {
  return GitHub;
};
