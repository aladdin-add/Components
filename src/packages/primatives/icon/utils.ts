import lazy from 'react-lazy-named';

export const getIcon = (config, name) => {
  if (config.iconDir) {
    switch (config.iconDir) {
      default:
        return lazy(() => import('react-feather'), name);
    }
  }

  throw new Error('Unknown icon lib');
};
