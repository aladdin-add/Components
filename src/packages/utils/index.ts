export * from './automation';
export * from './test-tools';
export * from './context/config';
export * from "./shouldForwardProp";