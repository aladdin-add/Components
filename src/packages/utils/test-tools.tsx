import React, { ReactNode } from 'react';
import { mount, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { ConfigContext } from '@/utils/context/config';
import theme from '@/theme';

const config = require('../../orchard.theme.config.json');

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: any) => {
      return (
        val &&
        typeof val === 'object' &&
        'props' in val &&
        Object.keys(val.props).some(prop => keys.some(key => key === prop))
      );
    },
    print: (val: any, serialize: any) => {
      keys.forEach(key => {
        delete val.props[key];
      });
      return serialize(val);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors.common,
    modes: theme.colors.modes,
    ...theme.colors.modes.light,
  },
};

export const shallowWithTheme = (children: ReactNode) =>
  shallow(<ThemeProvider theme={formattedTheme}>{children}</ThemeProvider>)
    .dive()
    .shallow();

export const mountWithTheme = (children: ReactNode) => {
  expect.addSnapshotSerializer(removeProperties());

  return mount(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>{children}</ThemeProvider>
    </ConfigContext.Provider>
  );
};
