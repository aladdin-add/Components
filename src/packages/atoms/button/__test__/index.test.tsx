import React from 'react';
import toJson from 'enzyme-to-json';
import { mountWithTheme } from '@/utils/test-tools';

import Button from '..';

describe('<Button />', () => {
  it('renders default correctly', () => {
    const wrapper = mountWithTheme(<Button>Default</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders primary correctly', () => {
    const wrapper = mountWithTheme(<Button variant="primary">Primary</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders secondary correctly', () => {
    const wrapper = mountWithTheme(
      <Button variant="secondary">Secondary</Button>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
