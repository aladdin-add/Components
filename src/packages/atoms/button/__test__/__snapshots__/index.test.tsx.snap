// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders default correctly 1`] = `
<Component>
  <Button>
    <Styled(Interactable)
      autoid="Default_button"
      forwardedAs="button"
      typography="button"
    >
      <Interactable
        as="button"
        autoid="Default_button"
        className="sc-hKFxyN eyTgNS"
        typography="button"
      >
        <styled.div
          as="button"
          className="sc-hKFxyN eyTgNS"
          data-autoid="default_button"
          onClick={[Function]}
          typography="button"
        >
          <button
            className="sc-dlnjwi bUluch sc-hKFxyN eyTgNS"
            data-autoid="default_button"
            onClick={[Function]}
          >
            <Text>
              <styled.span>
                <span
                  className="sc-bdnxRM eZabTl"
                >
                  Default
                </span>
              </styled.span>
            </Text>
          </button>
        </styled.div>
      </Interactable>
    </Styled(Interactable)>
  </Button>
</Component>
`;

exports[`<Button /> renders primary correctly 1`] = `
<Component>
  <Button
    variant="primary"
  >
    <Styled(Interactable)
      autoid="Primary_button"
      forwardedAs="button"
      typography="button"
      variant="primary"
    >
      <Interactable
        as="button"
        autoid="Primary_button"
        className="sc-hKFxyN bkQBjU"
        typography="button"
        variant="primary"
      >
        <styled.div
          as="button"
          className="sc-hKFxyN bkQBjU"
          data-autoid="primary_button"
          onClick={[Function]}
          typography="button"
          variant="primary"
        >
          <button
            className="sc-dlnjwi bUluch sc-hKFxyN bkQBjU"
            data-autoid="primary_button"
            onClick={[Function]}
          >
            <Text>
              <styled.span>
                <span
                  className="sc-bdnxRM eZabTl"
                >
                  Primary
                </span>
              </styled.span>
            </Text>
          </button>
        </styled.div>
      </Interactable>
    </Styled(Interactable)>
  </Button>
</Component>
`;

exports[`<Button /> renders secondary correctly 1`] = `
<Component>
  <Button
    variant="secondary"
  >
    <Styled(Interactable)
      autoid="Secondary_button"
      forwardedAs="button"
      typography="button"
      variant="secondary"
    >
      <Interactable
        as="button"
        autoid="Secondary_button"
        className="sc-hKFxyN bUpehk"
        typography="button"
        variant="secondary"
      >
        <styled.div
          as="button"
          className="sc-hKFxyN bUpehk"
          data-autoid="secondary_button"
          onClick={[Function]}
          typography="button"
          variant="secondary"
        >
          <button
            className="sc-dlnjwi bUluch sc-hKFxyN bUpehk"
            data-autoid="secondary_button"
            onClick={[Function]}
          >
            <Text>
              <styled.span>
                <span
                  className="sc-bdnxRM eZabTl"
                >
                  Secondary
                </span>
              </styled.span>
            </Text>
          </button>
        </styled.div>
      </Interactable>
    </Styled(Interactable)>
  </Button>
</Component>
`;
